api/v1alpha1/addon_types.go

```diff
@@ -134,40 +143,156 @@ func (ClusterAutoscaler) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
-type CSIProviders struct {
+type DefaultStorage struct {
+	ProviderName           string `json:"providerName"`
+	StorageClassConfigName string `json:"storageClassConfigName"`
+}
+
+type CSI struct {
 	// +optional
 	Providers []CSIProvider `json:"providers,omitempty"`
 	// +optional
-	DefaultClassName string `json:"defaultClassName,omitempty"`
+	DefaultStorage *DefaultStorage `json:"defaultStorage,omitempty"`
 }
 
 type CSIProvider struct {
-	Name string `json:"name,omitempty"`
+	Name string `json:"name"`
+
+	// +optional
+	StorageClassConfig []StorageClassConfig `json:"storageClassConfig,omitempty"`
+
+	Strategy AddonStrategy `json:"strategy"`
+
+	// +optional
+	Credentials *corev1.SecretReference `json:"credentials,omitempty"`
```

Comment:
How about we keep the Secret local to this namespace?
```suggestion
	Credentials *corev1.LocalObjectReference `json:"credentials,omitempty"`
```

Reply:
Any reason for doing this? This would be a secret in the default namespace wouldn't it?

Reply:
No it would be a secret in the namespace where the Cluster is.

>Any reason for doing this? 
Just for consistency with this, butcan come back to it in the future. https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/blob/47e578a14dc9855ab8f76dbfa087c20972fd41db/api/v1alpha1/clusterconfig_types.go#L253